python/private/py_twine.bzl

```diff
@@ -0,0 +1,106 @@
+"""Implementation of py_twine rule.
+
+Simply wraps the tool with a bash script and a runfiles manifest.
+See https://twine.readthedocs.io/
+"""
+
+load(":py_wheel.bzl", "PyWheelInfo")
+
+_attrs = {
+    "twine_bin": attr.label(
+        doc = """\
+A py_binary that runs the twine tool.
+The default value assumes you have `twine` listed in your own requirements.txt, and have run
+`pip_parse` with the default name of `pypi`.
+
+If these don't apply, you might use the `entry_point` helper to supply your own twine binary:
```

Comment:
This all seems rather brittle or convoluted...

Is this something people could instead specify during their WORKSPACE setup? e.g.
`rules_python_setup(twine="//my:twine")`? And if unspecified, we install or otherwise assume some default?

Is this something bzlmod could handle better than workspace? If so, maybe we should make this bzlmod-only?

This can also be where toolchains might help -- have a twine toolchain, then use the usual toolchain registration/setup to figure out how a user provides the twine executable.

Reply:
Another comment/question -- the requirements.txt this is referring to is the same one used for e.g. my py_binary's, right? Must my py_binary and twine share the same requirements.txt settings? I'm hoping that the particular build/version of twine is independent of _what_ it is publishing? Coupling the two seems like it would cause friction if twine's transitive deps aren't compatible with your py_binary's transitive deps.

Reply:
Yes that's right, users can install twine with a separate pip_parse rule.